File: TelerikUIExamples/TelerikUIExamples/DataFormCollapsibleGroups.m
//
//  DataFormCollapsableGroups.m
//  TelerikUIExamples
//
//  Copyright © 2015 Telerik. All rights reserved.
//

#import "DataFormCollapsibleGroups.h"
#import <TelerikUI/TelerikUI.h>
#import "EmployeeInfo.h"

@interface DataFormCollapsibleGroups () <TKDataFormDelegate>

@end

@implementation DataFormCollapsibleGroups
{
    TKDataForm *_dataForm;
    TKDataFormEntityDataSource *_dataSource;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    _dataForm = [[TKDataForm alloc] initWithFrame:self.view.bounds];
    _dataForm.autoresizingMask = UIViewAutoresizingFlexibleHeight | UIViewAutoresizingFlexibleWidth;
    _dataForm.backgroundColor = [UIColor colorWithRed:0.937 green:0.937 blue:0.960 alpha:1.0];
    _dataForm.delegate = self;
    [self.view addSubview:_dataForm];
    
    EmployeeInfo *info = [[EmployeeInfo alloc] init];
    _dataSource = [[TKDataFormEntityDataSource alloc] initWithObject:info];
    
    [_dataSource addGroupWithName:@"Personal Info" propertyNames:@[@"givenNames", @"surname", @"gender", @"idNumber", @"dateOfBirth"]];
    [_dataSource addGroupWithName:@"Contact Info" propertyNames:@[@"employeeId", @"phoneNumber"]];
    
    _dataSource[@"gender"].editorClass = [TKDataFormSegmentedEditor class];
    _dataSource[@"gender"].valuesProvider = @[@"Male", @"Female"];
    
    _dataSource[@"idNumber"].editorClass = [TKDataFormNumberEditor class];
    _dataSource[@"employeeId"].editorClass = [TKDataFormNumberEditor class];
    
    _dataForm.dataSource = _dataSource;
}

#pragma mark TKDataFormDelegate

- (void)dataForm:(TKDataForm *)dataForm updateGroupView:(TKEntityPropertyGroupView *)groupView forGroupAtIndex:(NSUInteger)groupIndex
{
    groupView.collapsible = YES;
    groupView.titleView.style.separatorColor = [TKSolidFill solidFillWithColor:[UIColor colorWithRed:0.784 green:0.780 blue:0.8 alpha:1.0]];
}

- (CGFloat)dataForm:(TKDataForm *)dataForm heightForHeaderInGroup:(NSUInteger)groupIndex
{
    return 44;
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/DataFormCollapsibleGroups.swift
//
//  DataFormCollapsableGroups.swift
//  TelerikUIExamplesInSwift
//
//  Copyright © 2015 Telerik. All rights reserved.
//

import UIKit

class DataFormCollapsibleGroups: UIViewController, TKDataFormDelegate {
    let dataSource = TKDataFormEntityDataSource()
    // >> dataform-obj-swift
    let dataForm = TKDataForm()
    // << dataform-obj-swift
    
    override func viewDidLoad() {
        super.viewDidLoad()
      // >> dataform-obj-swift
        dataForm.frame = self.view.bounds
        dataForm.delegate = self
        self.view.addSubview(dataForm)
        // << dataform-obj-swift
        dataForm.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        dataForm.backgroundColor = UIColor(red: 0.937, green: 0.937, blue: 0.960, alpha: 1.0)
    
        // >> dataform-obj-swift
        let info = EmployeeInfo()
        dataSource.sourceObject = info
        
        dataSource.addGroupWithName("Personal Info", propertyNames: ["givenNames", "surname", "gender", "idNumber", "dateOfBirth"])
        dataSource.addGroupWithName("Contact Info", propertyNames: ["employeeId", "phoneNumber"])
        
        dataSource["gender"].editorClass = TKDataFormSegmentedEditor.self
        dataSource["gender"].valuesProvider = ["Male", "Female"]
        
        dataSource["idNumber"].editorClass = TKDataFormNumberEditor.self
        dataSource["employeeId"].editorClass = TKDataFormNumberEditor.self
        dataSource["phoneNumber"].editorClass = TKDataFormPhoneEditor.self
       
        dataForm.dataSource = dataSource
        // << dataform-obj-swift
    }
    
    // >> dataform-collapse-swift
    func dataForm(dataForm: TKDataForm, updateGroupView groupView: TKEntityPropertyGroupView, forGroupAtIndex groupIndex: UInt) {
        groupView.collapsible = true
        groupView.titleView.style.separatorColor = TKSolidFill(color: UIColor(red: 0.784, green: 0.780, blue: 0.8, alpha: 1.0))
    }
    // << dataform-collapse-swift
    
    func dataForm(dataForm: TKDataForm, heightForHeaderInGroup groupIndex: UInt) -> CGFloat {
        return 44
    }
}
